### PR TITLE
New pedersen hash function

### PR DESCRIPTION
**Pull request by imeckler**
This replaces the existing pedersen hash function with the one described in the [zcash sapling spec](https://github.com/zcash/zips/blob/master/protocol/protocol.pdf), section 5.4.1.7.

The current one has the form `(t0, …, tn) -> t0 * g0 + … + tn * gn` where the `gi` are random group elements.

The new one has the form

```
(t0, …, t_{N-1}, t_N, …, t_{2N-1}, …)
->
t0 * (2^0 * g0)
+ t1 * (2^4 * g0)
+ t2 * (2^8 * g0)
+ …
+ t_{N-1} (2^{4*(N-1)} * g0)
+ tN * (2^{4 * N} * g1)
+ …
+ t_{2N-1} (2^{4*(2N - 1)} * g1)
+ …
```

There are several advantages of this hash function:
1. It is much faster to compute outside the SNARK.
2. The parameters are much smaller.

### CODE REVIEW

**Comment by psteckler** [comment]
Build error in unit tests:

```
File "lib/vrf_lib/integrated_test.ml", line 58, characters 19-22:
Error: Unbound record field acc
```

**Comment by psteckler** [comment]
Doc-comments need to be added to explain the purpose of each function.

File: src/lib/snark_params/snark_params.ml [hunk 1]
```diff
@@ -486,19 +497,91 @@ module Tick = struct
...
 
     let%test_unit "hash large number of chunks plus 2" =
-      For_tests.run_hash_test ((Chunked_triples.Chunk.size * 250) + 2)
+      For_tests.run_hash_test ((scalar_chunk_size * 10) + 2)
+*)
+
+    [%%if
+    not fake_hash]
+
+    (*
```
Comment (psteckler) [inline src/lib/snark_params/snark_params.ml]: why are these tests commented out?

**Comment by imeckler** [comment]
Closing this as it's actually slower than the old hash function and I realized I made an error in calculating the speed / memory-use ratio of this new version (it's worse than the old version).

**Pull request closed without merging**